cvmfs/bundle.cc

```diff
@@ -0,0 +1,87 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#include "bundle.h"
+
+#include <string>
+
+#include "json_document.h"
+#include "pack.h"
+#include "util/exception.h"
+#include "util/posix.h"
+#include "util/string.h"
+
+ObjectPack* Bundle::CreateBundle(const JSON* json_obj)
+{
+  const JSON* value = json_obj;
+  if (value->type != JSON_OBJECT) {
+    PANIC(kLogStderr, "JSON object not found");
+  } else {
+    // create an ObjectPack
+    ObjectPack* op = new ObjectPack();
```

Comment:
Let's use a `UniquePtr` from pointer.cc. We then don't need the explicit `delete` calls in the error paths. We return it with `UniquePtr::Release()`